src/components/modal/index.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useRef } from 'react';
+import './styles.css';
+
+const Modal = props => {
+  const modalRef = useRef();
+
+  useEffect(() => {
+    const clickOutsideContent = e => {
+      if (e.target === modalRef.current) {
+        props.setShow(false);
+      }
+    };
+    window.addEventListener('click', clickOutsideContent);
+    return () => {
+      window.removeEventListener('click', clickOutsideContent);
+    };
+  }, [props]);
+
+  return (
+    <div ref={modalRef} className={`modal ${props.show ? 'active' : ''}`}>
+      <div className="modal__content">{props.children}</div>
+    </div>
+  );
+};
+
+export default Modal;
+
+export const ModalHeader = props => {
+  return <div className="modal__header"> {props.children} </div>;
+};
+
+export const ModalBody = props => {
```

Comment:
If you're only using ModalBody maybe you could add it inside the return of `<div className="modal__content">` instead of exporting it on its own?